src/mem/cache/prefetch/stride.cc

```diff
@@ -133,7 +134,8 @@ Stride::calculatePrefetch(const PrefetchInfo &pfi,
     }
 
     // Get required packet info
-    Addr pf_addr = pfi.getAddr();
+    Addr pf_addr = useCachelineAddr ? blockAddress(pfi.getAddr())
```

Comment:
Is there a performance benefit on not using block addresses given that we end up rounding the prefetch strides to block addresses anyway?

Reply:
In a micro-benchmark that I wrote I saw a very small improvement in IPC when I was operating at the cache line granularity. This mainly happens because when we operate on word granularity, we may issue multiple prefetches for the same cache line for different words. Although these prefetches can be coalesced into one at the prefetch queue, we would still be exhausting the bandwidth of the prefetcher if we issue a lot of redundant prefetches. That's the reason why I've set the default parameters of the prefetcher such thatit operates on cache line granularity